transcoders.go

```diff
@@ -238,13 +238,66 @@ func garlic64BtS(b []byte) (string, error) {
 	return addr, nil
 }
 
-func garlicValidate(b []byte) error {
+func garlic64Validate(b []byte) error {
 	if len(b) < 386 {
 		return fmt.Errorf("failed to validate garlic addr: %s not an i2p base64 address. len: %d\n", b, len(b))
 	}
 	return nil
 }
 
+var TranscoderGarlic32 = NewTranscoderFromFunctions(garlic32StB, garlic32BtS, garlic32Validate)
+
+var garlicBase32Encoding = base32.NewEncoding("abcdefghijklmnopqrstuvwxyz234567")
+
+func garlic32StB(s string) ([]byte, error) {
+	s = strings.Replace(s, ".b32.i2p", "", -1)
```

Comment:
`strings.TrimSuffix`

But really, do we want to allow this? Personally, I wouldn't expect this suffix to be included in the multiaddr.